maven/src/main/java/org/commonjava/maven/galley/maven/model/view/MavenPomView.java

```diff
@@ -668,7 +672,13 @@ public ParentView getParent()
                 continue;
             }
 
-            result.add( new RepositoryView( node.getPomView(), node.getElement(), node.getOriginInfo() ) );
+            RepositoryView repoDeclaration =
```

Comment:
Actually, can you put this in a new method, called something like:

```
public List<RepositoryView> getNonProfileRepositories()
```

getAllRepositories implies that it will return any repo declaration from anywhere in the pom.xml...and this change alters that logic.